### PR TITLE
[WIP] Incorrect chainer err when test and after fail

### LINKED ISSUE #4062: Cypress commands (e.g. cy.get) return undefined if previous test step fails
### Current behavior:

Cypress test runner fails in `after()` block with error:

```
CypressError: cy.first() failed because it requires a DOM element.

The subject received was:

  > undefined

The previous command that ran was:

  > cy.get()

Because this error occurred during a 'after all' hook we are skipping the remaining tests in the current suite:
```

This happens each time a test step fails. In this case, it happens when the following assertion fails 
```js
.get('div.header__dob').should('contain', `${patient.birth_date}`)
```

### Desired behavior:

Cleanup step should execute with Cypress commands returning values instead of `undefined`

### Steps to reproduce: (app code and test code)

1. Write test with `before`, `test` & `after` steps
2. Have an assertion fail during testing
3. `cy.get` should return `undefined` in `after` step

### Versions

Cypress: 3.2.0
OS: MacOS 10.14.4
Browser: Electron59 & Chrome73

### PR DESCRIPTION
**Pull request by jennifer-shehane**
close #4062 

- [ ] fix error display
- [x] write test

### CODE REVIEW

File: packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_4062_spec.coffee [hunk 1]
```diff
@@ -0,0 +1,7 @@
+describe "issue 4062",  ->
+  afterEach ->
+    cy.get("element_does_not_exist", {timeout: 200}).first()
+
+  it "fails with correct chaingin afterEach err when both test and afterEach fail",  ->
```
Comment (TapaiBalazs) [inline packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_4062_spec.coffee]: I know it is not that important, but you have a typo: "chaingin".

Comment (jennifer-shehane) [reply packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_4062_spec.coffee]: 😝 thanks

**Comment by slava-44** [comment]
@jennifer-shehane do you have some workaround for this issue or any ideas when could it be fixed? I use hook 'after' very often in my tests and each time if some tests are failed this hook fails also.

**Comment by jennifer-shehane** [comment]
closing due to inactivity.

**Pull request closed without merging**